nrf52-hal-common/src/spim.rs

```diff
@@ -242,18 +268,14 @@ where
         chip_select: &mut Pin<Output<PushPull>>,
         buffer: &mut [u8],
     ) -> Result<(), Error> {
-        slice_in_ram_or(buffer, Error::DMABufferNotInDataMemory)?;
-
-        chip_select.set_low().unwrap();
-
-        // Don't return early, as we must reset the CS pin
-        let res = buffer.chunks(EASY_DMA_SIZE).try_for_each(|chunk| {
-            self.do_spi_dma_transfer(DmaSlice::from_slice(chunk), DmaSlice::from_slice(chunk))
-        });
+        let mut tx = self.transaction(chip_select);
+        let mut transfer = tx.transfer_polling(buffer)?;
 
-        chip_select.set_high().unwrap();
+        while !transfer.poll_complete()? {
+            spin_loop_hint();
+        }
```

Comment:
It might not hurt to add a method to the transfer types that does this loop, that way users can block on transfers if they want (eg. if the transfer was started in the background, but has to be completed before some operation can start).

Reply:
Yep, makes sense